### DISCUSSION TITLE
Define semi diameter for surface

### DISCUSSION THREAD

**Discussion by ppfromero** [mjhoptics/ray-optics#83] [Q&A] [answered]
Hello!

I'm beginner using ray-optics. 
I'm unable to define semi diameters for surfaces

I try 

`sm.add_surface([Radius, Thick, 'Mat, 'CAT', SD])`
but SD parameter is not take in account

Here below my system (doublet):

```
from rayoptics.environment import *

opm = OpticalModel()

sm = opm['seq_model']
osp = opm['optical_spec']
pm = opm['parax_model']
em = opm['ele_model']
pt = opm['part_tree']


osp['fov'] = FieldSpec(osp, key=['object', 'angle'], value=5, flds=[0., 0.7], is_relative=True)
osp['pupil'] = PupilSpec(osp, key=['object', 'pupil'], value=12.5)
osp['wvls'] = WvlSpec([('F', 0.5), (660, 1.0), ('C', 0.5)], ref_wl=1)

opm.system_spec.title = 'My design'
opm.system_spec.dimensions = 'MM'
opm.radius_mode = True
sm.gaps[0].thi=1e10


sm.add_surface([0., 10.])
sm.set_stop()
sm.add_surface([49.57, 7.0, 'H-ZF10', 'CDGM'])
sm.add_surface([-14.41, 1.97, 'H-LAK53B', 'CDGM'])
sm.add_surface([-44.94, 37.49])


opm.update_model()

sm.list_model()
pm.first_order_data()

layout_plt = plt.figure(FigureClass=InteractiveLayout, opt_model=opm).plot()
spot_plt = plt.figure(FigureClass=SpotDiagramFigure, opt_model=opm,scale_type=Fit.All_Same, dpi=200).plot()

opm.save_model("My_design")
```

I need to define something in order to take in account SD?

Thanks for your help

**Comment by mjhoptics** [comment, marked as answer]
Here's what I did to make it work, assuming sd=7 for the doublet..

```
sm.add_surface([0., 10., '', '', 6.25])
sm.set_stop()
sm.add_surface([49.57, 7.0, 'H-ZF10', 'CDGM', 7])
sm.add_surface([-14.41, 1.97, 'H-LAK53B', 'CDGM', 7])
sm.add_surface([-44.94, 37.49, '', '', 7])

sm.do_apertures = False

opm.update_model()
```
Note there is a flag, `do_apertures`, in the sequential model that automatically sets the surface clear apertures to pass the vignetted field rays. This will overwrite any apertures that have been entered (or 1.0, which is the default). Turning this off will take whatever apertures that have been entered (with 1.0 as a default). I don't like how hidden this is, but haven't been able to come up with a better approach yet.

Note that if there are no materials entered for the add_surface() call, you have to add empty strings to pad out the argument list to include the `sd`.

Hope this helps.
Mike

**Reply by ppfromero** [reply]
Super Mike I miss sm.do_apertures = False
Now it's working!!
